server/src/handlers/upload.rs

```diff
@@ -132,6 +109,33 @@ pub async fn upload_handler(
     )?))
 }
 
+
+async fn fs_upload(
```

Comment:
Maybe cleaner to have these as Methods on FileStore? Not sure if you agree. Can imagine it's more discoverable that way.

Reply:
Yes, still in progress. I just push a lot 🙂